### PR TITLE
Njs education page

### LINKED ISSUE #563: Shift Codebase to NextJS: Education Page
- Work with the `nextjs` branch. Do not disturb the `master` branch please
- Footer and Navbar should be called dynamically (for now don't do anything about it. It is a part of another issue)

### PR DESCRIPTION
**Pull request by ashuvssut**
closes #563 

nextjs education page done
-------------------

_--I have participated in DWoC SWoC and MWoC_

### CODE REVIEW

**Comment by ashuvssut** [comment]
closes #563

**Comment by anushbhatia** [comment]
Sorry, @ashuvssut I had to add navbar and rest of the components that's why modified the complete code.

**Comment by ashuvssut** [comment]
@anushbhatia plz review

File: package.json [hunk 1]
```diff
@@ -8,6 +8,14 @@
     "start": "next start"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^1.2.34",
+    "@fortawesome/free-brands-svg-icons": "^5.15.2",
+    "@fortawesome/free-solid-svg-icons": "^5.15.2",
+    "@fortawesome/react-fontawesome": "^0.1.14",
+    "@iconscout/unicons": "^3.0.6",
+    "@material-ui/core": "^4.11.3",
```
Comment (anushbhatia) [inline package.json:16]: What is this for

**Review by anushbhatia — CHANGES_REQUESTED** [review]
Don't add anything in public assets file as there are the specific file name components.
Add the page in the pages folder(that's fine)
See for the cards on experience page they are not.
Try making them with talwindcss

**Comment by ashuvssut** [comment]
@anushbhatia, please review home page PR #584 .  I will make changes and correct this one after that PR gets merged.

I am working on two separate issues. 
1. home page #557 
2. education page #563 

So I have made 2 separate PRs.

I know that education page has all the files required to make the home page... but this education page branch doesnt have all the new changes that home page branch has.

Moreover.. TBH😅...I want to get my both lvl5 PRs merged separately instead of just getting merged by a single lvl5 PR(this one)🤣🤣🤣🤣

File: package.json (hunk 1 above)
Comment (ashuvssut) [reply package.json:16]: the icons for Dribbble, Kaggle, Medium, google, etc are not available in material-ui/icons(already installed before)

so i installed fontawesome icons ....by installing @fortawesome package. It has all the free fontawesome icons.

The html-css-js version of this projects also used fontawesome. so I used also used it here

**Comment by ashuvssut** [comment]
> Don't add anything in public assets file as there are the specific file name components.
> Add the page in the pages folder(that's fine)
> See for the cards on experience page they are not.
> Try making them with talwindcss

@anushbhatia I have not added any page component in public folder. there are only images n svgs that are meant to be used in education page.

Also, in public/assets/js there are javascript files that  contain arrays like:- moocsCard array, etc...
there is also particle js file that is meant to be added in header of the rendered html.

But i have not added any component. If you see any page component there.... it is a useless code which I will remove eventually in my further nextjs PRs. I am determined to do all nextjs PRs.

And about tailwindcss... i kinda dont like bootstrap, tailwind, etc... I dont like remembering those classes. I am just the simple css user. I find it difficult to make responsive components using bootstrap, tailwind, etc...

**Comment by ashuvssut** [comment]
> @anushbhatia, please review home page PR #581 . I will make changes and correct this one after that PR gets merged.
> 
> I am working on two separate issues.
> 
> 1. home page #557
> 2. education page #563
> 
> So I have made 2 separate PRs.
> 
> I know that education page has all the files required to make the home page... but this education page branch doesnt have all the new changes that home page branch has.
> 
> Moreover.. TBH😅...I want to get my both lvl5 PRs merged separately instead of just getting merged by a single lvl5 PR(this one)🤣🤣🤣🤣

can you reopen the closed #581. I have pushed those new fixes to the error you reported

**Comment by anushbhatia** [comment]
Lib folder is there where you add the assets related which are basically scripts like particle js and other scripts that are specified in it. We don't want the complete same project being copy-pasted from the current project to a new one as that will result in better performance the current project has unnecessary scripts and code if we had to shift the complete project why would we need your help in that. We want the development in terms of the Nextjs and talwindcss and component should be created by that instead of this. So that extra css rendered can be eliminated that's the advantage of using NEXT and tailwind.

**Comment by anushbhatia** [comment]
Refer to these videos for references if you want https://react2025.com/

File: pages/index.js [hunk 2]
```diff
@@ -1,122 +1,108 @@
...
+import { faTwitter, faDribbble, faLinkedinIn, faKaggle, faMediumM, faGithub } from "@fortawesome/free-brands-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect } from "react";
+import ContactLinks from "./../components/contactlinks";
+import Container from "./../components/Navbar/Nav";
 
-var ReactRotatingText = require('react-rotating-text');
+var ReactRotatingText = require("react-rotating-text");
 export default function Home() {
+  useEffect(() => {
```
Comment (anushbhatia) [inline pages/index.js:11]: If you have modified the education page only why have you made the changes in the index.js that is irrelevant then

**Comment by anushbhatia** [comment]
> @anushbhatia, please review home page PR #581 . I will make changes and correct this one after that PR gets merged.
> 
> I am working on two separate issues.
> 
> 1. home page #557
> 2. education page #563
> 
> So I have made 2 separate PRs.
> 
> I know that education page has all the files required to make the home page... but this education page branch doesnt have all the new changes that home page branch has.
> 
> Moreover.. TBH😅...I want to get my both lvl5 PRs merged separately instead of just getting merged by a single lvl5 PR(this one)🤣🤣🤣🤣

Sure no problem will give points for both separately as you want.

**Comment by anushbhatia** [comment]
> > Don't add anything in public assets file as there are the specific file name components.
> > Add the page in the pages folder(that's fine)
> > See for the cards on experience page they are not.
> > Try making them with talwindcss
> 
> @anushbhatia I have not added any page component in public folder. there are only images n svgs that are meant to be used in education page.
> 
> Also, in public/assets/js there are javascript files that contain arrays like:- moocsCard array, etc...
> there is also particle js file that is meant to be added in header of the rendered html.
> 
> But i have not added any component. If you see any page component there.... it is a useless code which I will remove eventually in my further nextjs PRs. I am determined to do all nextjs PRs.
> 
> And about tailwindcss... i kinda dont like bootstrap, tailwind, etc... I dont like remembering those classes. I am just the simple css user. I find it difficult to make responsive components using bootstrap, tailwind, etc...

Let me discuss this with @smaranjitghose  and let you know if you are facing this issue then we can go stepwise first bootstrap then tailwind

**Comment by ashuvssut** [comment]
I believe that I can make responsive components by just simple css. If you want to use...tailwindcss instead.. that can be done by someone who likes to use ( having more experience) with tailwind. I can open up issues to use tailwindcss. Good contributors will surely contribute the tailwind code.

**Comment by ashuvssut** [comment]
> We don't want the complete same project being copy-pasted from the current project to a new one as that will result in better performance the current project has unnecessary scripts and code if we had to shift the complete project why would we need your help in that. We want the development in terms of the Nextjs and talwindcss and component should be created by that instead of this. So that extra css rendered can be eliminated that's the advantage of using NEXT and tailwind.

You said that you don't want the complete same project. So does this mean that I can modify the components stylings and add svgs/xtra relevant content etc wherever I feel good?

If yes then it u'd be great

**Comment by anushbhatia** [comment]
> > We don't want the complete same project being copy-pasted from the current project to a new one as that will result in better performance the current project has unnecessary scripts and code if we had to shift the complete project why would we need your help in that. We want the development in terms of the Nextjs and talwindcss and component should be created by that instead of this. So that extra css rendered can be eliminated that's the advantage of using NEXT and tailwind.
> 
> You said that you don't want the complete same project. So does this mean that I can modify the components stylings and add svgs/xtra relevant content etc wherever I feel good?
> 
> If yes then it u'd be great

Let me discuss this with him and this sounds great to me no problem

**Comment by anushbhatia** [comment]
@ashuvssut I have created a new branch for nextjs add all the pull request there. We are moving ahead to normal CSS to the talwindcss later

**Comment by ashuvssut** [comment]
amazing! thnx @anushbhatia 

learning nextjs is going to be amazing!

**Comment by anushbhatia** [comment]
i will create a base template and push

**Comment by ashuvssut** [comment]
Okay😁

I will be greateful if you could manage to create that branch before 24th Feb.

**Comment by anushbhatia** [comment]
> Okay😁
> 
> I will be greateful if you could manage to create that branch on or before 24th Feb.
> 
> If u can't manage to create that branch this soon...then it's also totally fine😄

Created it

**Comment by ashuvssut** [comment]
I will submit my work by tommorow

**Comment by smaranjitghose** [comment]
@ashuvssut Do attach some clips of the changes in the PR as well

**Pull request closed without merging**